test/test-template-engine.js

```diff
@@ -856,12 +857,14 @@ describe('rendering errors', () => {
       } catch (e) {
         error = e;
       }
-      assert(error?.message === `Found invalid template string "<div id="target" not-ok=" at " not-ok=".`, error.message);
+      assert(error?.message === `Found invalid template on or after line 1 in substring \`<div id="target" not-ok=\`. Failed to parse \` not-ok=\`.`, error.message);
       container.remove();
     });
 
+    // Note that this is _still_ line “1” because it begins with a newline. This
+    //  is what authors will expect as the “first line”.
```

Comment:
This is pretty nuanced, but I want the following to _both_ conceptually report the “first line”:

```js
html`<div .oops='${whoops}'></div>`;
```

```js
html`
  <div .oops='${whoops}'></div>
`;
```

… i.e., when the first line is _literally just a newline and nothing else_, ignore it, it’s not what I would consider the _first line_. The rational here is that if / when you could copy this text into a separate html file (not in our lifetimes), it would match the line numbering there.

Reply:
Interesting — IMO stripping the newlines is more confusing because the actual code is:
```

  <div .oops='${whoops}'></div>

```

I might prefer to keep this precise vs smart, but I do not feel strongly about it

Reply:
> stripping the newlines

🤔 — Hmm, definitely not stripping newlines! More like conditionally deleting the initial newline. I think you’d agree that in either case, the html file would look like this, right? I.e., the markup there _is on line one_. I can remove the fork though. Not worth arguing 🤙 

```html
<div></div>
```